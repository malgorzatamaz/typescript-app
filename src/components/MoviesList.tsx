import { Pagination, PaginationItem, PaginationLink } from "reactstrap";
import { range } from "lodash";
import React, { Dispatch, SetStateAction } from "react";
import styled from "styled-components";

import { Movie } from "types/movieTypes";
import margins from "assets/margins";
import MovieCard from "components/MovieCard";

export interface MoviesListProps {
  movies: Movie[];
  page: number;
  pagesCount: number;
  setPage: Dispatch<SetStateAction<number>>;
}
const StyledPagination = styled(Pagination)`
  margin-top: ${margins.m};
`;

const StyledList = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-top: ${margins.m};
  margin-bottom: ${margins.m};
`;

const MoviesList: React.FC<MoviesListProps> = ({
  movies,
  page,
  setPage,
  pagesCount,
}) => {
  const firstPage = page - 10 > 1 ? page - 10 : 1;
  const lastPage = page + 10 < pagesCount ? page + 10 : pagesCount;

  return (
    <React.Fragment>
      <StyledList>
        {movies.map((movie: Movie) => (
          <MovieCard key={movie.id} movie={movie} />
        ))}
      </StyledList>
      <StyledPagination>
        {range(firstPage, lastPage).map((pageNumber: number) => (
          <PaginationItem key={pageNumber} active={page === pageNumber}>
            <PaginationLink onClick={() => setPage(pageNumber)}>
              {pageNumber}
            </PaginationLink>
          </PaginationItem>
        ))}
      </StyledPagination>
    </React.Fragment>
  );
};

export default MoviesList;
