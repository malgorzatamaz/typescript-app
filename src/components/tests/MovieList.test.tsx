import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import MoviesList from "../MoviesList";

describe("MovieList", () => {
  const movies = [
    { title: "Title1", image: "img", id: "123", year: 2001 },
    { title: "Title2", image: "img", id: "124", year: 2012 },
  ];

  const props = {
    page: 1,
    setPage: jest.fn(),
    movies,
    pagesCount: 1,
  };

  test("renders movies list", () => {
    const element = render(
      <Router>
        <MoviesList {...props} />
      </Router>
    );
    expect(element).toMatchSnapshot();
  });

  test("list should have two elements", () => {
    render(
      <Router>
        <MoviesList {...props} />
      </Router>
    );
    const elem1 = screen.getByText("Title1");
    const elem2 = screen.getByText("Title2");

    expect(elem1).toBeInTheDocument();
    expect(elem2).toBeInTheDocument();
  });
});
