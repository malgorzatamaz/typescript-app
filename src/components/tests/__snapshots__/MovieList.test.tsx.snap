// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders movies list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlnjwi bsSfEf"
      >
        <div
          class="sc-bdnxRM bxAvtM card"
        >
          <img
            alt="Title1"
            src="img"
          />
          <div
            class="card-body"
          >
            <h6
              class="card-title"
            >
              Title1
            </h6>
            <p
              class="card-text"
            >
              2001
            </p>
            <a
              href="/movies/123"
            >
              Read more
            </a>
          </div>
        </div>
        <div
          class="sc-bdnxRM bxAvtM card"
        >
          <img
            alt="Title2"
            src="img"
          />
          <div
            class="card-body"
          >
            <h6
              class="card-title"
            >
              Title2
            </h6>
            <p
              class="card-text"
            >
              2012
            </p>
            <a
              href="/movies/124"
            >
              Read more
            </a>
          </div>
        </div>
      </div>
      <nav
        aria-label="pagination"
        class="sc-gtsrHT gzOPjC"
      >
        <ul
          class="pagination"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlnjwi bsSfEf"
    >
      <div
        class="sc-bdnxRM bxAvtM card"
      >
        <img
          alt="Title1"
          src="img"
        />
        <div
          class="card-body"
        >
          <h6
            class="card-title"
          >
            Title1
          </h6>
          <p
            class="card-text"
          >
            2001
          </p>
          <a
            href="/movies/123"
          >
            Read more
          </a>
        </div>
      </div>
      <div
        class="sc-bdnxRM bxAvtM card"
      >
        <img
          alt="Title2"
          src="img"
        />
        <div
          class="card-body"
        >
          <h6
            class="card-title"
          >
            Title2
          </h6>
          <p
            class="card-text"
          >
            2012
          </p>
          <a
            href="/movies/124"
          >
            Read more
          </a>
        </div>
      </div>
    </div>
    <nav
      aria-label="pagination"
      class="sc-gtsrHT gzOPjC"
    >
      <ul
        class="pagination"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
